src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -670,15 +672,65 @@ public void execute() throws GitException, InterruptedException {
     }
 
     /**
+     * clean_.
+     *
+     * @return a {@link org.jenkinsci.plugins.gitclient.CleanCommand} object.
+     */
+    public CleanCommand clean_() {
+        return new CleanCommand() {
+            private List<String> excludePatterns = Collections.emptyList();
+            private boolean removeSubmodules = false;
+            private Integer timeout = TIMEOUT;
+        	           
+            public CleanCommand excludePatterns(List<String> excludePatterns) {
+                this.excludePatterns = excludePatterns;
+                return this;
+            }
+
+            @Override
+            public CleanCommand submodules(boolean removeSubmodules) {
+                this.removeSubmodules = false;
+                return this;
+            }
+
+            public CleanCommand timeout(Integer timeout) {
+                this.timeout = timeout;
+                return this;
+            }
+            
+            public void execute() throws GitException, InterruptedException {
+                reset(true);
+                ArgumentListBuilder args = new ArgumentListBuilder();
+                args.add("clean");
+                if (removeSubmodules) {
+                    args.add("-ffdx");
+                }
+                else {
+                    args.add("-fdx");
+                }
+                if (excludePatterns != null) {
+                    for (String pattern : excludePatterns) {
+                        if ( !Strings.isNullOrEmpty(pattern) ) {
+                            args.add("-e");
+                            args.add(pattern);
+                        }
+                    }
+                }
+                String result = launchCommandIn(args, workspace, environment, timeout);
+                listener.getLogger().println(result);
+            }
+        };
+    }
+    
+    /**
      * Remove untracked files and directories, including files listed
      * in the ignore rules.
      *
      * @throws hudson.plugins.git.GitException if underlying git operation fails.
      * @throws java.lang.InterruptedException if interrupted.
      */
     public void clean() throws GitException, InterruptedException {
-        reset(true);
-        launchCommand("clean", "-fdx");
+    	clean_().execute();
```

Comment:
the reset is not removed, it is part of the execute method of the CleanCommand

ide puts in tabs, so that is not intentionally, should be fixed now